src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -346,7 +347,28 @@ public void completePayloadAttempt() {
             firstPayload = false;
           }
 
-          payload.append(PayloadBuilder.addContainerStartToOMFDataMessage(tagName));
+          String typeString = "";
+
+          if (tagPayload.getDataPoint().getType() == DataType.FLOAT) {
+            typeString = "number";
+          } else if (tagPayload.getDataPoint().getType() == DataType.INTEGER) {
+            typeString = "integer";
+          } else if (tagPayload.getDataPoint().getType() == DataType.BOOLEAN) {
+            typeString = "boolean";
+          } else if (tagPayload.getDataPoint().getType() == DataType.DWORD) {
+            typeString = "number";
+          } else if (tagPayload.getDataPoint().getType() == DataType.STRING) {
+            typeString = "string";
```

Comment:
magic string

Reply:
magic strings are always debatable but the fact that "string" shows up multiple times in this method makes it clear that it should be defined as a final variable.